R/arc-select.R

```diff
@@ -224,26 +233,27 @@ collect_layer <- function(
   # count the number of features in a query
   n_feats <- count_results(req, query_params)
 
-  if (is.null(n_feats)) {
-    cli::cli_abort(c(
-      "Cannot determine the number of features in request.",
-      "i" = "did you set custom parameters via {.arg ...}?"
-    ), call = error_call)
+  if (is.null(n_feats) && is.null(n_max)) {
+    cli::cli_abort(
+      c("Can't determine the number of features for {.arg x}.",
+        "*" = "Check to make sure your {.arg where} statement is valid or
+        set a value for {.arg n_max}."
+      ),
+      call = error_call
+    )
   }
 
   # identify the number of pages needed to return all features
   # if n_max is provided need to reduce the number of pages
   if (n_feats > n_max) {
-    n_feats <- n_max
-  }
-
-  if (is.null(n_feats)) {
-    cli::cli_abort(
-      c("Can't determine the number of features for {.arg x}.",
-        "*" = "Check to make sure your {.arg where} statement is valid."
-      ),
-      call = error_call
+    cli::cli_bullets(
+      c(
+        "i" = "Query results limited to {n_max} out of {n_feats} available feature{?s}.",
+        "!" = "Increase {.arg n_max} value to return all selected features."
+        )
```

Comment:
this is a verbose message which doesn't have any way to suppress the output. If we want to be extra loud, we should have a way to get rid of it. In `arcgisgeocode` I've added `.progress` arguments to include progress bars from `httr2::req_perform_parallel()` perhaps here we can use a `verbose = TRUE` default argument then conditionally provide this message.